lib/opam.ml

```diff
@@ -217,18 +245,25 @@ module Package_summary = struct
       Option.map ~f:OpamUrl.to_string (OpamFile.OPAM.dev_repo opam_file)
     in
     let depexts = OpamFile.OPAM.depexts opam_file in
-    { package; url_src; hashes; dev_repo; depexts }
+    let flags = OpamFile.OPAM.flags opam_file in
+    { package; url_src; hashes; dev_repo; depexts; flags }
+
+  let has_flag flag { flags; _ } = List.mem flag ~set:flags
+  let is_compiler v = has_flag OpamTypes.Pkgflag_Compiler v
 
   let is_virtual = function
     | { url_src = None; _ } -> true
     | { dev_repo = None | Some ""; _ } -> true
     | _ -> false
 
-  let is_base_package = function
-    | { package; _ }
-      when OpamPackage.Name.Set.mem package.name Config.base_packages ->
-        true
-    | _ -> false
+  let is_base_package v =
+    let in_base_pkgs { package; _ } =
```

Comment:
I find this naming confusing as "base package" now means both a package appearing in the list of base packages in the config (`in_base_pkgs`) and a package which appears in the base packages list or is a compiler packages (`is_base_package`).

Reply:
That's a good point. I've looked into this a bit and realized that all the checks always test for `is_base` and `is_virtual` at the same time so I created `is_safe` that calls `is_base`, `is_virtual` and also the new flag-based variants.

`is_safe_package` is not a great name, better suggestions welcome.